test/retry.js

```diff
@@ -71,6 +86,38 @@ test('falsy value prevents retries #2', async t => {
 	t.truthy(err);
 });
 
+test('custom retries', async t => {
+	let tried = false;
+	const err = await t.throws(got(`${s.url}/500`, {
+		throwHttpErrors: true,
+		retries: {
+			retry: iter => {
+				if (iter === 1) {
+					tried = true;
+					return 1;
+				}
+
+				return 0;
+			}, methods: [
+				'GET'
+			], statusCodes: [
+				500
+			]
+		}
+	}));
+	t.is(err.statusCode, 500);
+	t.true(tried);
+});
+
+test('respect 413 Retry-After', async t => {
+	const err = await got(`${s.url}/413`, {
+		throwHttpErrors: false,
+		retries: 1
+	});
+	t.is(err.statusCode, 413);
+	t.true(Number(err.body) >= 2000);
```

Comment:
Not immediately clear from reading the code what `2000` is. Make it a constant like `TOTAL_REQUEST_TIME` or something.